server/lib/nl/common/bad_words.py

```diff
@@ -81,7 +90,21 @@ def load_bad_words_file(local_file: str, validate: bool = False) -> BannedWords:
         continue
 
       # Generally, be resilient to duplicates.  Only assert when `validate` is set.
-      if _DELIM in line:
+      if _DELIM in line and _SET_START in line and _SET_END in line:
+        # In this case, split the line into _DELIM separated words.
+        line = line.replace(_SET_START, "").replace(_SET_END, "")
+        parts = line.split(_DELIM)
+        set_A = parts[0].split(",")
+        set_B = parts[1].split(",")
+
+        for w_a in set_A:
+          for w_b in set_B:
+            if w_a not in bad_words.entries:
```

Comment:
What do you think of skipping the {}, making this use the same logic as `cross` and generalizing to more than 2?

Somewhat like:

```
delimited_parts = [w.strip() for w in line.split(_DELIM) if w.strip()]
lists = [[x.strip() for x in p.split(',') if x.strip()] for p in delimited_parts]
for index_list in list(itertools.product(*lists)):
  words = sorted([lists[i][index] for i, index in enumerate(index_list)])
  bad_words.entries[words[0]].other_words.append(words[1:])
```

Reply:
Ah interesting. Sure, done!